Adding feature to ignore list and nuke errors (#114)
gruntwork-io/cloud-nuke: README.md

cloud-nuke aws --resource-type ec2 --dry-run
```

### Ignoring list/nuke errors for certain/all resources

Ideally you should have AWS IAM permissions to list and nuke all target resources. If you do not have permissions to
nuke resource X - you should exclude it using the `--exclude-resource-type` flag. However, if you don't know which resource

> types you are not permitted to nuke or you have list acceess for all but nuke excess for some, you can ignore their

Reviewer: ```suggestion
types you are not permitted to nuke or you have list access for all but nuke access for some, you can ignore their
```

Author: Fixed.

errors by specifying the `--ignore-errors` flag:

```shell
cloud-nuke aws --ignore-errors ec2
```

> This will ignore any errorrs encountered while listing/nuking EC2 resources - please note that if there are resources that

Reviewer: Please specify what other values you can pass here beyond `ec2` and how to handle multiple values.

Reviewer: Also, does this ignore an EC2 error and continue nuking EC2 instances... Or does it stop nuking EC2 instances at the first EC2 error, but then keeps nuking other types of resources? We should make the behavior clear in the docs to avoid surprises.

Author: Thanks for the feedback - clarified with example. It keeps on nuking other resource types - updated the README.

cloud-nuke aws --resource-type ec2 --dry-run
```

### Ignoring list/nuke errors for certain/all resources

Ideally you should have AWS IAM permissions to list and nuke all target resources. If you do not have permissions to

> nuke resource X - you should exclude it using the `--exclude-resource-type` flag. However, if you don't know which resource

Reviewer: ```suggestion
nuke resource X, you should exclude it using the `--exclude-resource-type` flag. However, if you don't know which resource
```

types you are not permitted to nuke or you have list access for all but nuke access for some, you can ignore their
errors by specifying the `--ignore-errors` flag:

```shell

> cloud-nuke aws --ignore-errors ec2 --ignore-errors s3

Reviewer: We still need to provide users with the a pointer to what resource types can be specified via `--ignore-errors`. i.e., It's more than just `ec2` and `s3`, right?

```

> This will ignore any errorrs encountered while listing/nuking EC2 and S3 resources. For example, if there are 3

Reviewer: ```suggestion
This will ignore any errors encountered while listing/nuking EC2 and S3 resources. For example, if there are 3
```

> EC2 instances and 2 S3 buckets to nuke and few/all of the EC2 instances and few/all of the S3 bucket deletion fails - cloud-nuke

Reviewer: ```suggestion
EC2 instances and 2 S3 buckets to nuke and few/all of the EC2 instances and few/all of the S3 bucket deletion fails, cloud-nuke
```

will log those errors and proceed with deleting further resources.

> Please note that cloud-nuke deletes resource types in a certain order to avoid dependency errors. If there are resources
> that should be deleted before EC2 or S3 - and there are errors encountered while listing/nuking those resources e.g.
> cloud-nuke deletes autoscaling groups before EC2 and if your role does not have permissions to delete autoscaling groups
> and you have not added `asg` to the `--ignore-errors` list, then cloud-nuke will error out and fail before reaching to EC2.

Reviewer: ```suggestion
Please note that cloud-nuke deletes resource types in a certain order to avoid dependency errors. If there are resources
that should be deleted before EC2 or S3, and there are errors encountered while listing/nuking those resources, then cloud-nuke will error out and fail before reaching to EC2 or S3. For example,
cloud-nuke deletes autoscaling groups before EC2 instances, and if your role does not have permissions to delete autoscaling groups
and you added `ec2` to the `--ignore-errors` list but not `asg`, then cloud-nuke will still end up exiting with an error around deleting those autoscaling groups.
```
